### PR TITLE
Finished new runup code

### PR DESCRIPTION
**Pull request by KateBrodie**
Created getRunupIndex
Integrated getRunupIndex into calculateRunupLine

### CODE REVIEW

File: Linescan/PreProcessing/calcRunupLine/calculateRunupLine.m [hunk 1]
```diff
@@ -12,9 +12,12 @@
...
 % - lineRunupCalc: [struct]: Calculated the runup line variables
 
-[lineRunupCalc.downLineIndex,lineRunupCalc.QAQCmeta.autoRunupCalcVersion]=getRunupIndx3(lineGriddedData.zGrid,lineCoredat.Hz);
-% lineRunupCalc.downLineIndex=round(50+20*sin((1:size(Rmat,1))/20));%make dummy index for now
-
+%Runup Parameters
+dx=diff(lineCoredat.downLineDistance(1:2));
+precision=0.015;
+filtseconds=30;
+%Digitize Runup
```
Comment (hokiespurs) [inline Linescan/PreProcessing/calcRunupLine/calculateRunupLine.m]: @KateBrodie Should these values come from the param file?

File: Linescan/PreProcessing/calcRunupLine/getRunupIndex.m [hunk 2]
```diff
@@ -1,15 +1,42 @@
...
+% filtseconds=30;
+% precision=0.015;
+
+Hz=round(Hz);
+
+%Tuneable parameters
 xShoreSmooth=4; %meters to smooth in the cross-shore
+xShoreSmooth2=2; %meters to smooth in the cross-shore and fill in nans
 timeSmooth=1.2; %seconds to smooth in time
 cumNormThresh=0.25; %percentage of time above minimum foreshore to be called water
```
Comment (hokiespurs) [inline Linescan/PreProcessing/calcRunupLine/getRunupIndex.m]: @KateBrodie should these come from a param as well?

**Comment by hokiespurs** [comment]
@KateBrodie @atdyer,

Code looks good! Just wondering about if we want those thresholds to be in our param file? Or hardcoded in.

**Comment by hokiespurs** [comment]
Also, @KateBrodie, thoughts on deleting the file "TEMPRUNUPSHITTHATKATENEEDS.m"?  Do you still need it?

**Comment by KateBrodie** [comment]
I wondered this too.

**Comment by KateBrodie** [comment]
Hmm I thought I deleted it? I meant to.

**Comment by KateBrodie** [comment]
My thought was maybe put the parameters like precision and filtlength in
the params file but the ones inside leave hard coded for now?

**Comment by hokiespurs** [comment]
@KateBrodie I went ahead and deleted that TEMPRUNUP... file

**Comment by hokiespurs** [comment]
@KateBrodie, what if they are all passed in as arguments to the function? Then we can have them all be in the param file, but you can also manually run and tweak things.

**Comment by KateBrodie** [comment]
ok @hokiespurs and @atdyer i made the thresholds variable inputs to both calculateRunupLine and getRunupIndex. They now reside in PreProcessLine...i guess they'll have to be added to the param file and that passed into PreProcessLine.

**Pull request merged**